### PR TITLE
pass registerer from storage to queue manager for its metrics

### PR DESCRIPTION
**Pull request by rfratto**
Continuing the discussion from #6726, this PR uses the previously unused registerer passed to NewStorage to create the metrics in the queue manager. 

Tests had to be modified to create new registerers so they didn't register the metrics twice.

Signed-off-by: Robert Fratto <robert.fratto@grafana.com>

/cc @roidelapluie @csmarchbanks

### CODE REVIEW

File: storage/remote/read_test.go [hunk 1]
```diff
@@ -93,7 +93,7 @@ func TestNoDuplicateReadConfigs(t *testing.T) {
 	}
 
 	for _, tc := range cases {
-		s := NewStorage(nil, prometheus.DefaultRegisterer, nil, dir, defaultFlushDeadline)
```
Comment (cstyan) [inline storage/remote/read_test.go]: was this change required for something?

Comment (rfratto) [reply storage/remote/read_test.go]: yeah, actually. `NewStorage` always calls `NewWriteStorage` even if there aren't any remote write configs, so the metrics were getting re-registered for subsequent iterations of the loop here

Comment (cstyan) [reply storage/remote/read_test.go]: Ah right I see now. This is another place you could just pass nil for the registerer if we don't care about any metric values during the test.

**Comment by rfratto** [comment]
@cstyan Everything passes when using nil for the registerer, PTAL
